Source/missiles.cpp

```diff
@@ -2593,20 +2580,20 @@ void AddFirewallC(MissileStruct &missile, Point dst, Direction midir)
 		int k = CrawlNum[i];
 		int ck = k + 2;
 		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
-			int tx = dst.x + CrawlTable[ck - 1];
-			int ty = dst.y + CrawlTable[ck];
-			if (0 < tx && tx < MAXDUNX && 0 < ty && ty < MAXDUNY) {
-				k = dPiece[tx][ty];
-				if (LineClearMissile(missile.position.start, { tx, ty })) {
-					if (missile.position.start != Point { tx, ty } && !nSolidTable[k] && dObject[tx][ty] == 0) {
-						missile.var1 = tx;
-						missile.var2 = ty;
-						missile.var5 = tx;
-						missile.var6 = ty;
-						missile._miDelFlag = false;
-						i = 6;
-						break;
-					}
+			Point target = dst + Displacement { CrawlTable[ck - 1], CrawlTable[ck] };
+			if (!InDungeonBounds(target))
+				continue;
+
+			k = dPiece[target.x][target.y];
+			if (LineClearMissile(missile.position.start, target)) {
+				if (missile.position.start != target && !nSolidTable[k] && dObject[target.x][target.y] == 0) {

```

Comment:
```suggestion
			if (LineClearMissile(missile.position.start, target)) {
				if (missile.position.start != target && IsTileNotSolid(target) && dObject[target.x][target.y] == 0) {
```
Using k as an alias doesn't do anything useful here.